app/helpers/hyrax/hyrax_helper_behavior.rb

```diff
@@ -54,6 +54,23 @@ def browser_supports_directory_upload?
       user_agent.include? 'Chrome'
     end
 
+    # @param [User] user
+    def render_notifications(user:)
+      mailbox = UserMailbox.new(user)
+      unread_notifications = mailbox.unread_count
+      link_to(hyrax.notifications_path,
+              'aria-label' => mailbox.label(params[:locale]),
```

Comment:
When the page is first rendered, before the actioncable subscriber connects, we don't know whether the user has just switched locales or not. So rather than deferring to the user's preferred locale, which hasn't been set since the *last* page load, we pass on the locale param from the current page load which has an overriding effect.